pkg/kubelet/secret/secret_manager.go

```diff
@@ -209,8 +209,12 @@ type cachingSecretManager struct {
 }
 
 func NewCachingSecretManager(kubeClient clientset.Interface) (Manager, error) {
+	// TODO: Currently apiserver completely cannot keep up with 1 minute ttl in
+	// 5000-node clusters. So before we optimize apiserver and/or etcd, increase
+	// the TTL to 2 minutes (which is a lot, but...)
+	ttl := 2 * time.Minute
```

Comment:
Agreed with @yujuhong above. Kubelet shouldn't make decision based on the cluster-wide knowledge. 

Also the proposed solution of listing all nodes at Kubelet initialization stage won't work since the cluster size might change: the new nodes might be added, and the old nodes might be removed at any given time, especially at the cluster creation time. Unless you want to Kubelet periodically list all nodes. This is not acceptable due to the cost applying to all the nodes in the cluster.